test/distributed/test_c10d_nccl.py

```diff
@@ -670,6 +670,19 @@ def test_non_blocking_with_eager_init(self):
         new_pg.broadcast(broadcast_tensor, 0).wait()
         self.assertEqual(backend.comm_split_count(), 1)
 
+    @requires_nccl()
+    @skip_but_pass_in_sandcastle_if(not TEST_MULTIGPU, "NCCL test requires 2+ GPUs")
+    def test_new_group_device_bound_local_sync_not_allowed(self):
+        store = c10d.FileStore(self.file_name, self.world_size)
+        device = torch.device(f"cuda:{self.rank}")
+        self._create_process_group_nccl(store, self.opts(), device_id=device)
+        with self.assertRaisesRegex(
+            ValueError,
+            "NCCL backend doesn't support use_local_synchronization=True "
+            "in sub-groups when device_id is used with init_process_group",
+        ):
```

Comment:
nit: shall we relax the matching condition or just expect a ValueError? That would be easier for a later developer if they happen need to change the error message.